daemon/container.go

```diff
@@ -1052,7 +1052,7 @@ func (container *Container) setupContainerDns() error {
 		return err
 	}
 
-	if config.NetworkMode != "host" {
+	if !config.NetworkMode.IsHost() || !config.NetworkMode.IsContainer() {
```

Comment:
my mistake here, This should be `!config.NetworkMode.IsHost() && !config.NetworkMode.IsContainer()` or `config.NetworkMode.IsBridge || config.NetworkMode.IsNone`, since only `--net=bridge` and `--net=none` has its
own private network stack at this time.
Do the checking in `runconfig.Parse` is to info the user
the setting for shared network mode is invalid or we have some 
docs to tell the user what they can set in shared net mode.